### PR TITLE
Add a duplicates_strategy argument to read_df…

### PR DESCRIPTION
**Pull request by ptormene**
…so in case of duplicates we can:
- keep the first occurrence,
- keep the last one
- calculate the average values

### CODE REVIEW

File: openquake/commonlib/readinput.py [hunk 1]
```diff
@@ -1066,21 +1066,52 @@ def get_exposure(oqparam, h5=None):
...
-    for duplicate sites, if any
+    Read a DataFrame containing lon-lat-id fields.
+    In case of rows having the same coordinates, duplicates_strategy
+    determines how to manage duplicates:
+
+    - 'error': raise an error (default)
+    - 'keep_first': keep the first occurrence
+    - 'keep_last': keep the last occurrence
+    - 'avg': calculate the average numeric values
     """
```
Comment (micheles) [inline openquake/commonlib/readinput.py:1086]: I would add a line
```python
assert duplicates_strategy in ('error', 'keep_first', 'keep_last', 'avg'), duplicates_strategy
```
just to be protected against mispellings.

File: openquake/commonlib/readinput.py [hunk 2]
```diff
@@ -1066,21 +1066,52 @@ def get_exposure(oqparam, h5=None):
     return exposure
 
 
-def read_df(fname, lon, lat, id):
+def concat_if_different(values):
+    unique_values = values.dropna().unique().astype(str)
+    # If all values are identical, return the single unique value,
+    # otherwise join with "|"
```
Comment (micheles) [inline openquake/commonlib/readinput.py:1072]: `return '|'.join(unique_values)` is shorter and does exactly the same thing.

File: openquake/calculators/base.py [hunk 3]
```diff
@@ -982,7 +982,8 @@ def _read_risk3(self):
             # NB: get_station_data is extending the complete sitecol
             # which then is associated to the site parameters below
             self.station_data, self.observed_imts = \
-                readinput.get_station_data(oq, self.sitecol)
+                readinput.get_station_data(oq, self.sitecol,
+                                           duplicates_strategy='error')
```
Comment (micheles) [inline openquake/calculators/base.py:986]: I would set `duplicates_strategy='avg'` so that the aristotle tests will get green.

**Comment by micheles** [comment]
LGTM, but please also update the changelog.

File: openquake/commonlib/readinput.py (hunk 2 above)
Comment (ptormene) [reply openquake/commonlib/readinput.py:1072]: Done

File: openquake/calculators/base.py (hunk 3 above)
Comment (ptormene) [reply openquake/calculators/base.py]: Done

**Pull request merged**